src/test/java/org/breedinginsight/brapps/importer/daos/BrAPIObservationUnitDAOTest.java

```diff
@@ -0,0 +1,144 @@
+
+import static org.breedinginsight.TestUtils.insertAndFetchTestProgram;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+@MicronautTest
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+@TestMethodOrder(MethodOrderer.OrderAnnotation.class)
+public class BrAPIObservationUnitDAOTest extends BrAPITest {
+
+    private FannyPack fp;
+
+    @Inject
+    private DSLContext dsl;
+
+    @Inject
+    private SpeciesDAO speciesDAO;
+
+    @Inject
+    private UserDAO userDAO;
+
+    @Inject
+    private BrAPIObservationUnitDAO obsUnitDAO;
+
+    @Inject
+    private ProgramService programService;
+
+    private Program validProgram;
+
+    private Gson gson;
+    private ImportUpload upload;
+
+    private BrAPIObservationTreatment testTreatment;
+
+    @Inject
+    @Client("/${micronaut.bi.api.version}")
+    RxHttpClient biClient;
+
+    @BeforeAll
+    @SneakyThrows
+    public void setup() {
+
+        ImportProgress progress = ImportProgress.builder().build();
+
+        upload = ImportUpload.uploadBuilder()
+                .progress(progress)
+                .build();
+
+        gson = new JSON().getGson();
+
+        // Add species needed to create program
+        fp = FannyPack.fill("src/test/resources/sql/brapi/species.sql");
+        super.getBrapiDsl().execute(fp.get("InsertSpecies"));
+        SpeciesEntity validSpecies = speciesDAO.findAll().get(0);
+
+        // Insert system admin role so can create program
+        FannyPack securityFp = FannyPack.fill("src/test/resources/sql/ProgramSecuredAnnotationRuleIntegrationTest.sql");
+        User testUser = userDAO.getUserByOrcId(TestTokenValidator.TEST_USER_ORCID).get();
+        dsl.execute(securityFp.get("InsertSystemRoleAdmin"), testUser.getId().toString());
+
+        SpeciesRequest speciesRequest = SpeciesRequest.builder()
+                .commonName(validSpecies.getCommonName())
+                .id(validSpecies.getId())
+                .build();
+
+        ProgramRequest program = ProgramRequest.builder()
+                .name("Test Program")
+                .species(speciesRequest)
+                .key("TEST")
+                .build();
+
+        // create test program
+        validProgram = insertAndFetchTestProgram(gson, biClient, program);
+        // updated with brapi db id
+        validProgram = programService.getById(validProgram.getId()).get();
+
+        testTreatment = new BrAPIObservationTreatment();
+        testTreatment.setFactor("ou1 treatment");
+    }
+
+    @Test
+    @SneakyThrows
+    @Order(1)
+    public void testCreateObservationUnitAdditionalInfoSingleTreatmentFactor() {
+        // create observation unit with treatments only in additional info to simulate breedbase not populating
+        // treatments field
+        BrAPIObservationUnit ou1 = new BrAPIObservationUnit();
+        ou1.setObservationUnitName("test1");
+        ou1.putAdditionalInfoItem(BrAPIAdditionalInfoFields.TREATMENTS, List.of(testTreatment));
```

Comment:
Should this test be updated to set the treatments in the `Treatments` field rather than in the `additionalInfo`?

Reply:
In this test I wanted to simulate what breedbase would do which would be to only have the treatments in the additionalinfo and not the treatments field.